### ISSUE TITLE
Replication of a remote db fails multiple times only if the "number of docs" and "update sequence" are different.

### ISSUE THREAD

**Issue by BroCube** [pouchdb/pouchdb#6152]
### Issue
When using the replicate function to replicate a remote database to a local one, PouchDB fails to connect and throws a generic error a random number of times, _if and only if_ the remote database has any revisions or deletions (i.e. if the number of docs and the update sequence are not exactly equal).

### Info
- Environment: Node.js on a new Ubuntu server for remote, standard html for the local instance
- Platform: Chrome
- Adapter: IndexedDB
- Server: CouchDB

### Reproduce
1. Create a remote database named `foo`. Create one document in that database. Update that document once.
2. Save this html and run it (be sure to update the` urlToRemoteDatabase` variable):
```
<script src="//cdn.jsdelivr.net/pouchdb/6.1.0/pouchdb.min.js"></script>
<script type="text/javascript">

/* put the URL of a remote database named "foo" here. */ 
var urlToRemoteDatabase = "http://example.com:5984/foo"

var foo = new PouchDB(urlToRemoteDatabase); 
var bar = new PouchDB('bar');

var timesFailed = 0;

function tryForever () {
	PouchDB.replicate(foo, bar)
	.on('active', function () {
	  console.log("Active. You see this every time tryForever() runs.");
	})
	.on('error', function (err) {
	  console.log("Error. You will see this every time you get ERR_CONNECTION_RESET");
	  console.log(err);
	  timesFailed++;
	  tryForever();
	})
	.on('change', function (info) {
		console.log("Change. You will see this once, after the program finally connects and sees there are new docs to write");
	})
	.on('paused', function (err) {
	  console.log("Paused. You will see this once, after success, when all docs are written and the connection closes.");
	})
	.on('complete', function (info) {
	  console.log("Complete. You will see this once, after all docs are written.");
	  console.log("I failed " + timesFailed + " times before I finally succeeded");
	})
}

//run it, and check your console
tryForever();
</script>
```

3. try this a few times, **being sure to clear your cache and delete your IndexedDB** using the chrome dev tool **before each attempt**.

Following these steps, you will see 

`POST http://example.com:5984/foo/_bulk_get?revs=true&latest=true net::ERR_CONNECTION_RESET`

a random number of times (sometimes zero, most I have seen is eleven). If you log the error, you will see a generic error

`message : "batch processing terminated with error"`

![error object](https://cloud.githubusercontent.com/assets/10625887/22125419/78c1a3a4-de48-11e6-9598-7eaf7de1e1a8.png)

This error looks similar to https://github.com/pouchdb/pouchdb/issues/5810, but it is a completely different issue. The above code will work completely fine without errors ever, if your remote `foo` database's update sequence and number of documents is exactly equal, regardless of the amount of docs, or the data in them (e.g. if `foo` has 3 docs, and the update sequence is 3, the above code will connect perfectly on the first attempt every time). Only when the sequence number and document quantity do not align does this error occur (e.g. from deleting or revising a doc).

**Comment by willholley** [comment]
Just to add some context, the code path that is taken for `rev-1` documents is different - the description implies that the issue occurs when any document exists with more than one revision, resulting in PouchDB using `_bulk_get` if the server supports it or falling back to individual document fetches.

I think this is addressed by #6036 - can you see whether upgrading to PouchDB 6.1.1 fixes your issue?

**Comment by BroCube** [comment]
By my testing it seems to have done so, yes! Thank you.
